src/test/java/com/amazon/ion/bytecode/bin11/bytearray/ReferenceOpcodeHandlerTest.kt

```diff
@@ -2,145 +2,38 @@
-                // encounter during parsing.
-                val payload = ByteArray(payloadLength)
-                arguments.add(
-                    Arguments.of(
-                        ReferenceOpcodeHandler(instruction),
-                        byteArrayOf(
-                            opcode.toByte(), // write the opcode
-                            *flexUIntStr.trim().hexStringToByteArray(), // then the FlexUInt
-                            *payload // then the payload bytes
-                        ),
-                        intArrayOf(
-                            instruction.packInstructionData(payloadLength),
-                            expectedPayloadStartPosition
-                        ),
-                        expectedEndPosition
-                    )
-                )
-            }
-        }
-
-        return arguments
-    }
-
     /**
      * Test that variable-length payload opcodes generate the correct *_REF bytecode.
      * Does not validate the actual payload in any way.
      */
     @ParameterizedTest
-    @MethodSource("referenceOpcodeHandlerTestCases")
-    fun `handlers for OP_X_REF opcodes emit correct bytecode`(
-        handler: ReferenceOpcodeHandler,
-        inputByteArray: ByteArray,
-        expectedBytecode: IntArray,
-        expectedEndPosition: Int
-    ) {
-        val buffer = BytecodeBuffer()
-
-        var position = 0
-        val opcode = inputByteArray[position++].unsignedToInt()
-        position += handler.convertOpcodeToBytecode(
-            opcode,
-            inputByteArray,
-            position,
-            buffer,
-            ConstantPool(0),
-            intArrayOf(),
-            intArrayOf(),
-            arrayOf()
-        )
-
-        assertEqualBytecode(expectedBytecode, buffer.toArray())
-        assertEquals(expectedEndPosition, position)
+    @MethodSource(REFERENCE_OPCODE_CASES)
+    fun `handlers for OP_X_REF opcodes emit correct bytecode`(input: String, bytecode: String) {
+        val opcode = input.take(2).toInt(16)
+        val handler = when (opcode) {
+            OpCode.ANNOTATION_TEXT -> ReferenceOpcodeHandler(Instructions.I_ANNOTATION_REF)
+            OpCode.VARIABLE_LENGTH_INTEGER -> ReferenceOpcodeHandler(Instructions.I_INT_REF)
+            OpCode.VARIABLE_LENGTH_DECIMAL -> ReferenceOpcodeHandler(Instructions.I_DECIMAL_REF)
+            OpCode.VARIABLE_LENGTH_TIMESTAMP -> ReferenceOpcodeHandler(Instructions.I_TIMESTAMP_REF)
+            OpCode.VARIABLE_LENGTH_STRING -> ReferenceOpcodeHandler(Instructions.I_STRING_REF)
+            OpCode.VARIABLE_LENGTH_SYMBOL -> ReferenceOpcodeHandler(Instructions.I_SYMBOL_REF)
+            OpCode.VARIABLE_LENGTH_BLOB -> ReferenceOpcodeHandler(Instructions.I_BLOB_REF)
+            OpCode.VARIABLE_LENGTH_CLOB -> ReferenceOpcodeHandler(Instructions.I_CLOB_REF)
+            else -> fail("Opcode is not a variable-length reference opcode: 0x${opcode.toSingleHexByte()}")
```

Comment:
Or you could just use the standard library functions.
```suggestion
            else -> fail("Opcode is not a variable-length reference opcode: 0x${opcode.toByte().toHexString()}")
```